### DISCUSSION TITLE
Adding Labels to Contour Lines

### DISCUSSION THREAD

**Discussion by AlexanderJuestel** [pyvista/pyvista#3882] [Q&A] [answered]
### Describe the feature you would like to be added.

Contour lines of meshes can easily be created using `mesh.contour(np.arange(start, stop, step))`. An additional improvement would be to label these contour lines as they may represent depths of a geological surface for instance (as a result of the modeling with GemPy).

So, apart from the color coding of the mesh, the labeling would provide useful information on the depths of the surface :) 

If you use `p.add_point_labels()` on the contour lines, you will get a label at every point, which is not very useful :)

### Pseudocode or Screenshots

![image](https://user-images.githubusercontent.com/45469915/213164704-68d8b067-2c78-4c27-a009-a8c72a250357.png)

**Comment by banesullivan** [comment, marked as answer]
Maybe something like this where I grab a single point from each contour line:


```py
import pyvista as pv
from pyvista import examples
import numpy as np

from pyvista._vtk import vtkMapper
vtkMapper.SetResolveCoincidentTopologyToPolygonOffset()
vtkMapper.SetResolveCoincidentTopologyPolygonOffsetParameters(0.1, 0.1)

mesh = examples.load_random_hills()
cntrs = mesh.contour(5)

levels = cntrs.split_bodies()
pts = []
values = []
for level in levels:
    pt = level.points[0]
    v = level['Elevation'][0]
    pts.append(pt)
    values.append(v)

pl = pv.Plotter()
pl.add_mesh(mesh, cmap='gist_earth')
pl.add_mesh(cntrs, color='white', line_width=3)
pl.add_point_labels(np.array(pts), [f'{v:.1f}' for v in values])
pl.show()
```
![screenshot](https://user-images.githubusercontent.com/22067021/214641537-5f6484e3-c449-420e-9e83-86caf9dbaf4a.png)

**Reply by AlexanderJuestel** [reply]
That looks like a very smooth solution! Thanks @banesullivan

**Reply by AlexanderJuestel** [reply]
It needs a little adaption (due to my data) but it works nicely!

![image](https://user-images.githubusercontent.com/45469915/215032209-89fb698c-c2c7-4548-bea5-4fe8adcd6aec.png)
